src/sagemaker_pytorch_container/deep_learning_container.py

```diff
@@ -0,0 +1,65 @@
+import json
+import logging
+import requests
+
+
+def _retrieve_instance_id():
+    """
+    Retrieve instance ID from instance metadata service
+    """
+    url = "https://169.254.169.254/latest/meta-data/instance-id"
+    response = requests_helper(url, timeout=0.01)
```

Comment:
Do we know the P99 response time of meta-data endpoints?

Reply:
script name - is a business decision - keeping it as it is for now
will check the p99 response time - if there is an API that supports calculating it - see documentation for details on the times